lib/blockscore/base.rb

```diff
@@ -1,10 +1,12 @@
 require 'blockscore/connection'
+require 'uri'
 
 module BlockScore
   class Base
     extend Connection
 
     ABSTRACT_WARNING = 'Base is an abstract class, not an API resource'.freeze
+    API_URL = URI.parse('https://api.blockscore.com/').freeze
```

Comment:
> The more dependencies on other gems the more possibility also of issues with Ruby version support conflicts

This is a valid concern.

> After I looked at Addressable::URI I saw no distinct advantage.

It handles more valid kinds of URIs than the normal URI class. The URI class has been basically frozen in time for the last 5-10 years and imho has been poorly maintained, as has much of ruby's core/stdlib. There are in-the-wild URIs that it cannot handle. The addressable gem more closely adheres to modern RFCs.

The API is mostly compatible, so refactoring to it later would not be much of an issue. If we run into actual issues that a modern URI replacement could resolve then we can revisit this.